src/store/emptyImageBuilderApi.ts

```diff
@@ -5,6 +5,24 @@ import { IMAGE_BUILDER_API } from '../constants';
 // initialize an empty api service that we'll inject endpoints into later as needed
 export const emptyImageBuilderApi = createApi({
   reducerPath: 'imageBuilderApi',
-  baseQuery: fetchBaseQuery({ baseUrl: IMAGE_BUILDER_API }),
+  baseQuery: fetchBaseQuery({
+    baseUrl: IMAGE_BUILDER_API,
+    paramsSerializer: (params) => {
+      const searchParams = new URLSearchParams();
+      for (const [key, value] of Object.entries(params)) {
+        if (value === undefined) {
+          // avoid sending undefined parameters
+          break;
```

Comment:
`break` will exit the loop immediately, which is not what we want if the first element has an undefined value, because the others after that might be valid and must be included. see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/break

`continue` on the other hand just jumps to the next loop. see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/continue

What I would do, if using `continue` is not what you want, is simply to change the logic of the if conditions:

```
if (value){
  if (Array.isArray(value)){
  }else{
  }
}
```